libs/server/Resp/Objects/SortedSetCommands.cs

```diff
@@ -911,5 +912,287 @@ private unsafe bool SortedSetDifferenceStore<TGarnetApi>(ref TGarnetApi storageA
 
             return true;
         }
+
+        /// <summary>
+        /// Computes an intersection operation between multiple sorted sets
+        /// and returns the result to the client.
+        /// </summary>
+        /// <param name="storageApi"></param>
+        /// <returns></returns>
+        private unsafe bool SortedSetIntersect<TGarnetApi>(ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            if (parseState.Count < 2)
+            {
+                return AbortWithWrongNumberOfArguments(nameof(RespCommand.ZINTER));
+            }
+
+            // Number of keys
+            if (!parseState.TryGetInt(0, out var nKeys))
+            {
+                return AbortWithErrorMessage(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER);
+            }
+
+            if (nKeys < 1)
+            {
+                return AbortWithErrorMessage(Encoding.ASCII.GetBytes(string.Format(CmdStrings.GenericErrAtLeastOneKey, "zinter")));
+            }
+
+            if (parseState.Count < nKeys + 1)
+            {
+                return AbortWithErrorMessage(CmdStrings.RESP_SYNTAX_ERROR);
+            }
+
+            var includeWithScores = false;
+            double[] weights = null;
+            var aggregateType = SortedSetAggregateType.Sum;
+            var currentArg = nKeys + 1;
+
+            // Read all the keys 
+            var keys = parseState.Parameters.Slice(1, nKeys);
+
+            // Parse optional arguments
+            while (currentArg < parseState.Count)
+            {
+                var arg = parseState.GetArgSliceByRef(currentArg).ReadOnlySpan;
+
+                if (arg.EqualsUpperCaseSpanIgnoringCase(CmdStrings.WITHSCORES))
+                {
+                    includeWithScores = true;
+                    currentArg++;
+                }
+                else if (arg.EqualsUpperCaseSpanIgnoringCase(CmdStrings.WEIGHTS))
+                {
+                    currentArg++;
+                    if (currentArg + nKeys > parseState.Count)
+                    {
+                        return AbortWithErrorMessage(CmdStrings.RESP_SYNTAX_ERROR);
+                    }
+
+                    weights = new double[nKeys];
+                    for (var i = 0; i < nKeys; i++)
+                    {
+                        if (!parseState.TryGetDouble(currentArg + i, out weights[i]))
+                        {
+                            return AbortWithErrorMessage(Encoding.ASCII.GetBytes(string.Format(CmdStrings.GenericErrNotAFloat, "weight")));
+                        }
+                    }
+                    currentArg += nKeys;
+                }
+                else if (arg.EqualsUpperCaseSpanIgnoringCase(CmdStrings.AGGREGATE))
+                {
+                    currentArg++;
+                    if (currentArg >= parseState.Count)
```

Comment:
nit: can join these two 'if's with an '||', just for better readability.

Reply:
Done